### PR TITLE
Add Auto EEID Assignment in FlexiblePatternGenerator

### LINKED ISSUE #29529: Enable Auto EEID Assignment in FlexiblePatternGenerator
### Motivation

`FlexiblePatternGenerator` currently does not support auto EEID assignment for its unit input meshes, making some data analyses difficult.

### Design

Provide an option to automatically assign an EEID to each unit input mesh with user-specified EEID name and shift.

### Impact

Better user experience.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #29529

### CODE REVIEW

File: modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C [hunk 1]
```diff
@@ -151,6 +171,8 @@ FlexiblePatternGenerator::validParams()
       "boundary_type boundary_mesh boundary_sectors boundary_size "
       "delete_default_external_boundary_from_inputs external_boundary_id external_boundary_name",
       "Boundary");
+  params.addParamNamesToGroup("cell_id_name cell_id_shift pattern_id_name pattern_id_shift",
+                              "Reporting Id Parameters");
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C:175]:
```suggestion
                              "Reporting Id");
```
Parameters gets added automatically at the end already

File: modules/reactor/test/tests/meshgenerators/flexible_pattern_generator/fp_eeid_pattern.i [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    num_sectors = 9
+    ring_radii = '2'
+    ring_intervals = '1'
+    ring_block_ids = '10'
+    ring_block_names = 'accg_1'
+    create_outward_interface_boundaries = false
+  []
+  [fpg]
+    type = FlexiblePatternGenerator
+    inputs = 'accg_1'
```
Comment (GiudGiud) [inline modules/reactor/test/tests/meshgenerators/flexible_pattern_generator/fp_eeid_pattern.i:13]:
```suggestion
    inputs = 'accg_1 accg_2'
```
let's add another one here so you get a non-uniform pattern id

**Review by GiudGiud — COMMENTED** [review]
looks good

**Comment by miaoyinb** [comment]
Thanks for the review. I modified the test to have two input meshes.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**